File: app/src/test/java/com/mgaetan89/showsrage/network/SickRageApi_GetCredentialsTest.java
package com.mgaetan89.showsrage.network;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class SickRageApi_GetCredentialsTest {
	@Parameterized.Parameter(3)
	public String credentials;

	@Parameterized.Parameter(2)
	public String password;

	@Parameterized.Parameter(0)
	public boolean useBasicAuthentication;

	@Parameterized.Parameter(1)
	public String username;

	@Test
	public void getCredentials() {
		assertThat(SickRageApi.getCredentials(this.useBasicAuthentication, this.username, this.password)).isEqualTo(this.credentials);
	}

	@Parameterized.Parameters(name = "{index}: {0} [{1}:{2}]")
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{false, null, null, null},
				{false, null, "", null},
				{false, null, "password", null},
				{false, "", "password", null},
				{false, "username", "password", null},
				{false, "username", "", null},
				{false, "username", null, null},
				{true, null, null, null},
				{true, null, "", null},
				{true, null, "password", null},
				{true, "", "password", null},
				{true, "username", "password", "Basic dXNlcm5hbWU6cGFzc3dvcmQ="},
				{true, "username", "", null},
				{true, "username", null, null},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/network/SickRageApi_GetCredentialsTest.kt
package com.mgaetan89.showsrage.network

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class SickRageApi_GetCredentialsTest(val useBasicAuthentication: Boolean, val username: String?, val password: String?, val credentials: String?) {
    @Test
    fun getCredentials() {
        assertThat(SickRageApi.getCredentials(this.useBasicAuthentication, this.username, this.password)).isEqualTo(this.credentials)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters(name = "{index}: {0} [{1}:{2}]")
        fun data(): Collection<Array<Any?>> {
            return listOf(
                    arrayOf<Any?>(false, null, null, null),
                    arrayOf(false, null, "", null),
                    arrayOf(false, null, "password", null),
                    arrayOf(false, "", "password", null),
                    arrayOf(false, "username", "password", null),
                    arrayOf(false, "username", "", null),
                    arrayOf(false, "username", null, null),
                    arrayOf<Any?>(true, null, null, null),
                    arrayOf(true, null, "", null),
                    arrayOf(true, null, "password", null),
                    arrayOf(true, "", "password", null),
                    arrayOf<Any?>(true, "username", "password", "Basic dXNlcm5hbWU6cGFzc3dvcmQ="),
                    arrayOf(true, "username", "", null),
                    arrayOf(true, "username", null, null)
            )
        }
    }
}
